File: Algorithms/scala/linearSearch.scala
object Script{
    def RecLinearSearch(li: List[Int], target : Int):Boolean  = {
        def helper(n:Int):Boolean = {
            if (li(n) == target) true
            else if (n == 0) false
            else helper(n-1)
    }
        helper(li.length-1)
    }
    def main(args: Array[String]): Unit = {
        val li = List(1,2,3,4,5)
        val target = 3
        val result = RecLinearSearch(li, target)
        println(result)
    }
}

File: Algorithms/python/linearSearch.py
# Implementacão Imperativa
def LinearSearch(li,target):
    for item in li:
        if item == target:
            return True
    return False

# Implementacão Funcional
def RecLinearSearch(li,target):
    def helper(n):
        if li[n] == target:
            return True
        elif n == 0:
            return False
        else:
            return helper(n-1)
    return  helper(len(li)-1)




# Testing setup - - - - - - - - - - - - - - - - - - - - -
test_list = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17]

print(LinearSearch(test_list, 12))
print(RecLinearSearch(test_list, 12))

print(LinearSearch(test_list, 20))
print(RecLinearSearch(test_list, 20))
# - - - - - - - - - - - - - - - - - - - - - -

